src/main/java/megamind/task/Deadline.java

```diff
@@ -0,0 +1,38 @@
+package megamind.task;
+
+import java.io.Serial;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+
+/**
+ * The `Deadline` class represents a task with a specific deadline.
+ * It extends the `Task` class and includes additional information about the deadline of the task.
+ * The class provides methods to construct a deadline task and return its string representation.
+ */
+public class Deadline extends Task {
```

Comment:
I like how you separated the tasks into different files instead of clumping it all into one, makes the code cleaner.